src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsSnapshotTaker.java

```diff
@@ -25,8 +26,13 @@ public StandardUsernamePasswordCredentials snapshot(StandardUsernamePasswordCred
         if (credentials instanceof UsernamePasswordCredentialsImpl) {
             return credentials;
         }
-        UsernamePasswordCredentialsImpl snapshot = new UsernamePasswordCredentialsImpl(credentials.getScope(), credentials.getId(), credentials.getDescription(), credentials.getUsername(), Secret.toString(credentials.getPassword()));
-        snapshot.setUsernameSecret(credentials.isUsernameSecret());
-        return snapshot;
+        try {
+            UsernamePasswordCredentialsImpl snapshot =
+                    new UsernamePasswordCredentialsImpl(credentials.getScope(), credentials.getId(), credentials.getDescription(), credentials.getUsername(), Secret.toString(credentials.getPassword()));
+            snapshot.setUsernameSecret(credentials.isUsernameSecret());
+            return snapshot;
+        } catch (Descriptor.FormException e) {
+            throw new RuntimeException(e);
```

Comment:
weired.  I was expecting the method to throw be able to throw an `IOException` as many credentials can (and are) documented to do that.  [doc](https://github.com/jenkinsci/credentials-plugin/blob/master/docs/implementation.adoc#:~:text=You%20should%20note%20that%20we%20have%20used%20Secret%20to%20define%20the%20secret.%20And%20furthermore%2C%20we%20have%20added%20the%20throws%20IOException%2C%20InterruptedException%20to%20the%20getter!%20Again%20this%20is%20so%20that%20a%20new%20CredentialsProvider%20can%20lazily%20fetch%20the%20secret.)